src/worker-blockprocessor.ts

```diff
@@ -23,136 +23,122 @@ process
     process.exit(1);
   });
 
-let jayson = require("jayson/promise");
-let rpc = url.parse(process.env.BITCOIN_RPC);
-let client = jayson.client.http(rpc);
-
-const LAST_PROCESSED_BLOCK = "LAST_PROCESSED_BLOCK";
-
-async function processBlock(blockNum, sendQueueRepository: Repository<SendQueue>) {
-  console.log("processing new block", +blockNum);
-  const responseGetblockhash = await client.request("getblockhash", [blockNum]);
-  const responseGetblock = await client.request("getblock", [responseGetblockhash.result, 2]);
-  const addresses: string[] = [];
-  const allPotentialPushPayloadsArray: Components.Schemas.PushNotificationOnchainAddressGotPaid[] = [];
-  const txids: string[] = [];
-  for (const tx of responseGetblock.result.tx) {
-    txids.push(tx.txid);
-    if (tx.vout) {
-      for (const output of tx.vout) {
-        if (output.scriptPubKey && output.scriptPubKey.addresses) {
-          for (const address of output.scriptPubKey.addresses) {
-            addresses.push(address);
-            const payload: Components.Schemas.PushNotificationOnchainAddressGotPaid = {
-              address,
-              txid: tx.txid,
-              sat: Math.floor(output.value * 100000000),
-              type: 2,
-              level: "transactions",
-              token: "",
-              os: "ios",
-            };
-            allPotentialPushPayloadsArray.push(payload);
-          }
-        }
-      }
+const groupByKey = <T>(list: T[], key: string) => list.reduce((hash, obj: any) => (
+  { ...hash, [obj[key]]: (hash[obj[key]] || []).concat(obj) }), {} as {[key: string]: T[]});
+
+async function processTransactions(transactions: Transaction[], sendQueueRepository: Repository<SendQueue>) {
+  // checking if there is a subscription to one of the mined txids:
+  const includedTransactionIds = transactions.map((tx) => tx.transactionId);
+  if (includedTransactionIds.length) {
+    const query2 = ds.getRepository(TokenToTxid).createQueryBuilder().where("txid IN (:...txids)", { txids: includedTransactionIds });
+    for (const t2txid of await query2.getMany()) {
+      const payload: Components.Schemas.PushNotificationTxidGotConfirmed = {
+        txid: t2txid.txid,
+        type: 4,
+        level: "transactions",
+        token: t2txid.token,
+        os: t2txid.os === "ios" ? "ios" : "android",
+        badge: 1,
```

Comment:
maybe its time to add `chain` to openapi definition of `Components.Schemas.PushNotificationTxidGotConfirmed`..?